cobbler/utils/threading.py

```diff
@@ -0,0 +1,97 @@
+        """
+        This constructor creates a Cobbler thread which then may be run by calling ``run()``.
+
+        :param event_id: The event-id which is associated with this thread. Also used as thread name
+        :param remote: The Cobbler remote object to execute actions with.
+        :param options: Additional options which can be passed into the Thread.
+        :param task_name: The high level task name which is used to trigger pre- and post-task triggers
+        :param api: The Cobbler api object to resolve information with.
+        """
+        Thread.__init__(self, name=event_id)
+        self.event_id = event_id
+        self.remote = remote
+        self.logger = logging.getLogger()
+        self.__setup_logger()
+        if options is None:
+            options = {}
+        self.options = options
+        self.task_name = task_name
+        self.api = api
+
+    def __setup_logger(self):
+        """
+        Utility function that will setup the Python logger for the tasks in a special directory.
+        """
+        filename = pathlib.Path("/var/log/cobbler/tasks") / f"{self.event_id}.log"
+        task_log_handler = logging.FileHandler(str(filename), encoding="utf-8")
+        task_log_formatter = logging.Formatter(
+            "[%(threadName)s] %(asctime)s - %(levelname)s | %(message)s"
+        )
+        task_log_handler.setFormatter(task_log_formatter)
+        self.logger.setLevel(logging.INFO)
+        self.logger.addHandler(task_log_handler)
+
+    def on_done(self):
+        """
+        This stub is needed to satisfy the Python inheritance chain.
+        """
+
+    def run(self):
+        """
+        Run the thread.
+
+        :return: The return code of the action. This may possibly a boolean or a Linux return code.
+        """
+        self.logger.info("start_task(%s); event_id(%s)", self.task_name, self.event_id)
+        time.sleep(1)
+        try:
+            if utils.run_triggers(
+                self.api,
+                None,
+                f"/var/lib/cobbler/triggers/task/{self.task_name}/pre/*",
+                self.options,
+            ):
+                self.remote._set_task_state(
+                    self, self.event_id, enums.EventStatus.FAILED
+                )
+                return False
+            rc = self._run(self)
+            if rc is not None and not rc:
+                self.remote._set_task_state(
+                    self, self.event_id, enums.EventStatus.FAILED
+                )
+            else:
+                self.remote._set_task_state(
+                    self, self.event_id, enums.EventStatus.COMPLETE
+                )
+                self.on_done()
+                utils.run_triggers(
+                    self.api,
+                    None,
+                    f"/var/lib/cobbler/triggers/task/{self.task_name}/post/*",
+                    self.options,
+                )
+            return rc
+        except:
```

Comment:
Best thing I can do here is to add `Exception` since this can be any exception thrown by the task that is being executed. Is this good enough in your eyes or what is your proposal?